webanno-project/src/test/java/de/tudarmstadt/ukp/clarin/webanno/project/ProjectServiceImplTest.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.ProjectService;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.model.ProjectPermission;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.Role;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.User;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class ProjectServiceImplTest
+{
+    private ProjectService sut;
+
+    @Autowired
+    private TestEntityManager testEntityManager;
+        
+    private Project testProject;
+    private User beate;
+    private User kevin;
+    
+    @Before
+    public void setUp() throws Exception
+    {
+        sut = new ProjectServiceImpl(testEntityManager.getEntityManager());
+        
+        //create users
+        beate = new User("beate", Role.ROLE_USER, Role.ROLE_ADMIN);
+        kevin = new User("kevin", Role.ROLE_USER);
+        User noPermissionUser = new User("noPermission", Role.ROLE_USER);
+        testEntityManager.persist(beate);
+        testEntityManager.persist(noPermissionUser);
+        testEntityManager.persist(kevin);
+        
+        //create project and projectPermissions for users
+        testProject = new Project("testProject");
+        testEntityManager.persist(testProject);
+        testEntityManager.persist(new ProjectPermission(testProject, "beate", ANNOTATOR));
+        testEntityManager.persist(new ProjectPermission(testProject, "kevin", ANNOTATOR));
+        testEntityManager.persist(new ProjectPermission(testProject, "beate", CURATOR));
+    }
+    
+    @After
+    public void tearDown() {
+        testEntityManager.clear();
+    }
+    
+    @SpringBootConfiguration
+    @EnableAutoConfiguration 
+    @EntityScan(
+            basePackages = {
+                "de.tudarmstadt.ukp.inception.curation",
+                "de.tudarmstadt.ukp.clarin.webanno.model",
+                "de.tudarmstadt.ukp.clarin.webanno.security.model" 
+    })
+    public static class SpringConfig {
+        // No content
+    }
+    
+    @Test
+    public void listProjectUsersWithPermissions_ShouldReturnUsers() {
+        List<User> foundUsers = sut.listProjectUsersWithPermissions(testProject);
+        
+        assertThat(foundUsers).hasSize(2);
+        assertThat(foundUsers).contains(beate, kevin);
+    }
+    
+    @Test
+    public void listProjectUsersWithSpecificPermissions_ShouldReturnUsers() {
+        List<User> foundUsers = sut.listProjectUsersWithPermissions(testProject, ANNOTATOR);
+        
+        assertThat(foundUsers).hasSize(2);
```

Comment:
how about instead of checking for size and containment to use `containsExactly`?